File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.js
function solution(X, Y) {
    let Xarr = X.split("").sort();
    let Yarr = Y.split("").sort();
    
    let i = 0;
    let j = 0;
    let answer = [];
    
    while(i < Xarr.length && j < Yarr.length){
        if(Xarr[i] == Yarr[j]){
            answer.push(Xarr[i]);
            i++;
            j++;
        }else if(Xarr[i] < Yarr[j]){
            i++;
        }else{
            j++;
        }
    }
    
    if(answer.length === 0){return "-1";}
    
    answer.sort((a,b)=>b-a);
    
    if(answer[0] === "0"){return "0";}
    
    return answer.join("");
}

File: 프로그래머스/1/131128. 숫자 짝꿍/숫자 짝꿍.py
def solution(X, Y):
    answer = ''
    Xarr = {}
    
    for i in X:
        Xarr[i] = Xarr.get(i,0)+1 
    
    for y in Y:
        if Xarr.get(y,0) > 0:
            answer+=y
            Xarr[y] -= 1
            
    answer = ''.join(sorted(answer,reverse=True))
    if answer == "": return "-1"
    if answer[0] == '0': return "0"

    return answer
    
